### PR TITLE
[IOTDB-6031] Fix SonarCloud bugs in confignode module

### PR DESCRIPTION
**Pull request by BUAAserein**
#### Fix some bugs in confignode module.

### CODE REVIEW

**Review by OneSizeFitsQuorum — COMMENTED** [review]
PTAL

File: iotdb-core/confignode/src/main/java/org/apache/iotdb/confignode/procedure/Procedure.java [hunk 1]
```diff
@@ -63,7 +64,7 @@ public abstract class Procedure<Env> implements Comparable<Procedure<Env>> {
   private volatile long timeout = NO_TIMEOUT;
   private volatile long lastUpdate;
 
-  private volatile byte[] result = null;
+  private AtomicReference<byte[]> result = null;
```
[the remark this answers is not in the document]
Comment (BUAAserein) [reply iotdb-core/confignode/src/main/java/org/apache/iotdb/confignode/procedure/Procedure.java:67]: AtomicReferenceArray requires Byte type, which introduces additional type cast overhead

[the remark this answers is not in the document]
Comment (OneSizeFitsQuorum) [reply iotdb-core/confignode/src/main/java/org/apache/iotdb/confignode/procedure/Procedure.java:67]: I see

**Review by OneSizeFitsQuorum — APPROVED** [review]
LGTM

**Pull request merged**